pages/attacks/Embedding_Null_Code.md

```diff
@@ -14,65 +14,53 @@ permalink: /attacks/Embedding_Null_Code
-technique to perform other attacks, like directory browsing, path
-traversal, SQL injection, execution of arbitrary code, among others. It
-can be found lots of vulnerable applications and exploits available to
-abuse systems using this technique.
+technique to perform other attack such as directory browsing, path
+traversal, SQL injection, execution of arbitrary code, and others. It
+can be found in lots of vulnerable applications and there are lots of exploits
+available to abuse systems.
 
 This technique includes several variations to represent the postfix NULL
 terminator:
 
-`PATH%00`
-`PATH[0x00]`
-`PATH[alternate representation of NULL character]`
+`PATH%00`  
+`PATH[0x00]`  
+`PATH[alternate representation of NULL character]`  
+`%00`  
 
-<script>
+## Examples
 
-</script>
+### PHP Script
 
-%00
+The following example shows the use of this technique to modify
+a URL and access arbitrary files on a filesystem due a PHP script
+vulnerability.
 
-## Example
-
-### Example1 – PHP Script
-
-In the following example, it’s shown the use of this technique to modify
-an URL and access arbitrary files on a filesystem due a vulnerability on
-PHP script.
-
-`$whatever = addslashes($_REQUEST['whatever']);`
+`$whatever = addslashes($_REQUEST['whatever']);`  
 `include("/path/to/program/" . $whatever . "/header.htm");`
 
-By manipulating the URL using postfix NULL bytes, one can have access to
+By manipulating the URL using postfix NULL bytes, one can access the
 UNIX password file:
 
 `http://vuln.example.com/phpscript.php?whatever=../../../etc/passwd%00`
 
-### Example2 – Adobe PDF ActiveX Attack
-
-Another know attack, consists in the exploitation of buffer overflow in
-the ActiveX component (pdf.ocx) that allows remote execution arbitrary
-code.
-
-The problem happens when a link is requested as:
+### Adobe PDF ActiveX Attack
 
-`GET /some_dir/file.pdf.pdf%00[long string] HTTP/1.1`
+Another attacks consists of exploitating buffer overflow in ActiveX components
+ (pdf.ocx) to allow remote code execution.
 
-In this case, the request must be made to a web server that truncates
-the request at the null byte (%00), as Microsoft IIS and Netscape
-Enterprise web servers. Though the requested URI is truncated for the
-purposes of locating the file, the long string is still passed to the
-Adobe ActiveX component. This component triggers a buffers overflow
-within RTLHeapFree() allowing for an attacker to overwrite an arbitrary
-word in memory.
+The attack occurs  when a link is requested as follows:
 
-This attack can be performed by adding malicious content to the end of
-any embedded link and referencing any Microsoft IIS or Netscape
-Enterprise web server. It is not necessary to establish a malicious web
-site to execute this attack.
+`GET /some_dir/file.pdf.pdf%00[long string] HTTP/1.1`  
 
-For more details about this attack, see External References topics.
+This exploit can be used on against a web server that truncates the request at 
```

Comment:
‘on’ or ‘against’ not both

Reply:
I don't actually know if I screwed up or if that was in the original text, but fixed.